kuma/users/signal_handlers.py

```diff
@@ -1,23 +1,67 @@
-from allauth.account.signals import email_confirmed, user_signed_up
+from allauth.account.signals import (
+    email_confirmed,
+    user_logged_in,
+    user_signed_up)
+from allauth.socialaccount.signals import pre_social_login
 from allauth.socialaccount.signals import social_account_removed
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import transaction
 from django.db.models.signals import post_delete, post_save, pre_delete
 from django.dispatch import receiver
 from waffle import switch_is_active
 
+from kuma.core.ga_tracking import (
+    ACTION_AUTH_SUCCESSFUL,
+    ACTION_FREE_NEWSLETTER,
+    ACTION_PROFILE_ALREADY_CREATED,
+    ACTION_PROFILE_CREATED,
+    CATEGORY_SIGNUP_FLOW,
+    track_event)
 from kuma.payments.utils import cancel_stripe_customer_subscription
 from kuma.wiki.jobs import DocumentContributorsJob
 
 from .models import User, UserBan
 from .tasks import send_welcome_email
 
 
+@receiver(pre_social_login, dispatch_uid='users.pre_social_login')
+def on_pre_social_login(sender, request, **kwargs):
+    """
+    Signal handler to be called when a given user has at least successfully
+    authenticated with a provider but not necessarily fully logged in on
+    our site. For example, if the user hasn't created a profile (e.g.
+    agreeing to terms and conditions) the 'user_logged_in' won't fire until
+    then.
+    """
+    sociallogin = kwargs.get('sociallogin')
+    if sociallogin:
+        track_event(
+            CATEGORY_SIGNUP_FLOW,
+            ACTION_AUTH_SUCCESSFUL,
+            sociallogin.account.provider)
+
+
 @receiver(user_signed_up, dispatch_uid='users.user_signed_up')
 def on_user_signed_up(sender, request, user, **kwargs):
     """
     Signal handler to be called when a given user has signed up.
     """
```

Comment:
Here we need to add something like:
```python
if getattr(request, 'social_auth_added', False):
    return
```
since we don't want to do any of the things we do in this signal handler (i.e., sending GA events for `profile-created` and `free-newsletter` as well as sending the welcome email) when the user is simply adding another social provider to their existing MDN account.